protocol.md

```diff
@@ -870,6 +902,13 @@ export interface TextDocumentClientCapabilities {
 ```typescript
 interface ClientCapabilities {
 	/**
+	 * The client supports $/partialResult notifications.
+	 * If true, the server is allowed to let the result in the final response be null
+	 * and only send the result through $/partialResult
+	 */
+	streaming?: boolean;
```

Comment:
> If a client and server adds support for streaming for one method, do they get streaming for all methods for free?

That's how I see it. If you implement streaming in the server for example, you will probably do it on the "dispatcher" level. So your individual handlers would return a result before, but can now take a progress callback, or also return an Observable instead of a Promise. If a method does not actually yield incremental results, the dispatcher can just transform the return value to a single replace operation.
For the client it's simple too, just implement applying JSON patches. A method that does not make use of it yet would just apply all patches and then wait until the final response comes in before rendering the UI.

I actually though about putting the `streaming` field into all the `RegistrationOptions` for every method, but I think that would make it more complex in the end because every client and server would have to look for what to do for every single method.